spec/lib/companies_selector_spec.rb

```diff
@@ -0,0 +1,129 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'CompaniesSelector' do
+  context 'with query params set (condition default)' do
+    it 'select companies accroding to the params' do
+      Company.create!([
+        {
+          name: 'MERCATOR',
+          isin: 'PLMRCTR00015',
+          stock: 'GPW',
+          condition: 'company_active',
+        },
+        {
+          name: 'MOSTALPLC',
+          isin: 'PLMSTPL00018',
+          stock: 'NC',
+          condition: 'company_active',
+        },
+      ])
+
+      params = ActionController::Parameters.new(query: 'MOS')
```

Comment:
In my opinion the code in `lib` should not know that you are processing params from the HTTP request (from Rails controller). The `CompaniesSelector` should expect to get an argument that is a hash or have an equivalent interface (you can pass `ActionController::Parameters` there only because it behaves like a hash). So here, in the for `CompaniesSelector` I would use a hash:

```suggestion
      params = {query: 'MOS'}
```

You will be able to pass `ActionController::Parameters` to `CompaniesSelector` as well, because `ActionController::Parameters` behaves like a hash, but you should not care about it in this test. This test should be interface-agnostic (when you are writing this test, imagine you will be using this class from Ruby console or some other random interface).

The same is true for the rest of the file.

Reply:
Ok, I understand what you mean, so then I should pass `params` to the `CompaniesSelector` instead of `params.permit(...)`. `permit` is not a hash method. But here I'm testing `CompaniesSelector` and I pass `params` with value set by me in the test, not by the user, so it looks good.